Wangscape/noise/module/Bitmap.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Bitmap.h"
+#include "logging/Logging.h"
+
+namespace noise
+{
+namespace module
+{
+
+namespace {
+// TODO (Serin): extract the data format from noise::RasterBase into a separate class and use it here
+static std::map<std::string, std::shared_ptr<sf::Image>> loadedImages;
+}
+
+
+Bitmap::Bitmap() : 
+    Module::Module(GetSourceModuleCount())
+{
+}
+
+const std::string & Bitmap::GetFilename() const
+{
+    return mFilename;
+}
+
+void Bitmap::SetFilename(const std::string & filename)
+{
+    // Load and process file if not already done
+    auto it = loadedImages.find(filename);
+    if (it == loadedImages.cend())
+        return;
+    else
```

Comment:
There are cases where an unnecessary `else` is clearer, but you're right that this is probably not one of then.